users/signals.py

```diff
@@ -0,0 +1,27 @@
+from django.db.models.signals import post_save, pre_save
+from django.dispatch import receiver
+from .models import User, Profile, UserSettings
+from .helpers import nickname_from_email
+
+
+# Do I need other signals?
+
+@receiver(post_save, sender=User)
+def create_profile(sender, instance, created, **kwargs):
+    if created:
+        profile = Profile.objects.create()
+        if instance.profile and instance.nickname:
+            nickname = instance.profile.nickname
+        else:
+            nickname = nickname_from_email(User, instance)
```

Comment:
No. Reddit has this required upon registration, so should we.

Reply:
There is no nickname creation from email any more